velox/exec/StreamingAggregation.cpp

```diff
@@ -275,14 +266,24 @@ RowVectorPtr StreamingAggregation::getOutput() {
   newGroups.resize(numGroups_ - numPrevGroups);
   std::iota(newGroups.begin(), newGroups.end(), numPrevGroups);
 
-  for (auto i = 0; i < aggregates_.size(); ++i) {
-    auto& aggregate = aggregates_.at(i).function;
+  for (const auto& aggregate : aggregates_) {
+    if (!aggregate.sortingKeys.empty()) {
+      continue;
+    }
+    auto& function = aggregate.function;
 
-    aggregate->initializeNewGroups(
+    function->initializeNewGroups(
         groups_.data(), folly::Range(newGroups.data(), newGroups.size()));
   }
 
-  evaluateAggregates();
+  if (sortedAggregations_) {
+    if (!newGroups.empty()) {
```

Comment:
Should we put this check around the loop over aggregates_ as well?

Reply:
Added, and I put all the new group init into a new function initializeNewGroups.